### PR TITLE
Dataparser image flags

### PR DESCRIPTION
**Pull request by khuston**
@richardjgowers  Fixes ignored image flags in dataparser.

Changes made in this Pull Request:
 - Restore original logic for getting positions from datafile

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/topology/LAMMPSParser.py [hunk 1]
```diff
@@ -256,8 +256,12 @@ def _parse_pos(self, datalines, pos):
         """Strip coordinate info into np array"""
         for line in datalines:
             line = line.split()
+            n = len(line)
             idx = int(line[0]) - 1
-            x, y, z = map(float, line[-3:])
+            if n in (7, 10):
+                x, y, z = map(float, line[4:7])
+            elif n in (6, 9):
+                x, y, z = map(float, line[3:6])
```
Comment (richardjgowers) [inline package/MDAnalysis/topology/LAMMPSParser.py]: It's quicker just to do `pos[idx] = line[3:6]/line[4:7]` and let numpy figure it out than cast to float ourselves

Comment (kain88-de) [inline package/MDAnalysis/topology/LAMMPSParser.py]: Also `np.float32` accepts list so no need for map. And another thing please prefer list comprehension to map. It does the same but people are more likely to have seem the comprehension

**Comment by khuston** [comment]
@richardjgowers I followed your simplification.

Sorry about the messy commit history. Had I realized those commits from late July/early August would show up, I would've re-forked.

**Comment by kain88-de** [comment]
If you rebase against the source branch next time instead of merging it with your feature branch you can clean up the git history. Right now with the merge I would advice against that.

**Comment by richardjgowers** [comment]
Thanks @khuston

**Pull request merged**